nmdc_schema/migration_recursion.py

```diff
@@ -283,11 +292,114 @@ def __init__(self) -> None:
 
     def force_research_study_study_category(self, study: dict) -> dict:
         if 'study_category' not in study:
-            logger.info(f"Forcing 'study_category: research_study' on {study['id']}")
+            logger.info(
+                f"Forcing 'study_category: research_study' on {study['id']}")
             study['study_category'] = 'research_study'
         return study
 
 
+class Migrator_from_8_1_0_to_9_0_0(MigratorBase):
+    """Migrates data from schema 8.1.0 to 9.0.0"""
+
+    def __init__(self) -> None:
+        """Invokes parent constructor and populates collection-to-transformations map."""
+
+        super().__init__()
+
+        # Populate the "collection-to-transformers" map for this specific migration.
+        self.agenda = dict(
+            study_set=[self.fix_award_dois, self.fix_pub_dois,
+                       self.fix_massive, self.fix_ess_dive, self.remove_doi_slots],
+        )
+
+    def process_doi(self, study: dict, doi_list, doi_category):
```

Comment:
If you end up making additional commits to this branch, will you add type hints to the `doi_list` and `doi_category` parameters? 

If they are a list and a string, the type hints would be:
```diff
- def process_doi(self, study: dict, doi_list, doi_category):
+ def process_doi(self, study: dict, doi_list: list, doi_category: str):
```

In color:
```py
def process_doi(self, study: dict, doi_list: list, doi_category: str):
```

Reference: List of [built-in types](https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html#useful-built-in-types) in the mypy documentation